generator/vertx/src/main/java/io/fabric8/maven/generator/vertx/PortsExtractor.java

```diff
@@ -0,0 +1,132 @@
+package io.fabric8.maven.generator.vertx;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+
+import org.apache.maven.model.Plugin;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import io.fabric8.maven.core.util.PrefixedLogger;
+import io.fabric8.utils.Strings;
+
+import static io.fabric8.maven.generator.vertx.Constants.*;
+
+public class PortsExtractor {
```

Comment:
If we can get rid of vertx deps, we could put this into a core util package to allow port extraction from arbitrary configuration files. This might be very helpful for other generators, too. Wonder whether to put this into the `core` or `generator.api`.